planner/core/plan_cache.go

```diff
@@ -533,7 +533,7 @@ func rebuildRange(p Plan) error {
 					if err != nil {
 						return err
 					}
-					if len(ranges) != len(x.Handles) && !isSafeRange(x.AccessConditions, &ranger.DetachRangeResult{
+					if len(ranges) != len(x.Handles) || !isSafeRange(x.AccessConditions, &ranger.DetachRangeResult{
```

Comment:
Here we add more strict restrictions for safety?

Reply:
Yes

Reply:
Here is an example about range length, when the first execution, we got [[1], [2]] from where k in (1, 2), and when the second execution, we got [[1]] from where k in (1, 1). For safety, if the range lengths changed, we generate a new plan instead of re-using the old one.